components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/WorkflowsApiServiceImpl.java

```diff
@@ -78,7 +78,14 @@ public Response workflowsUpdateWorkflowStatusPost(String workflowReferenceId, Wo
                 }
 
                 if (body.getStatus() == null) {
-                    RestApiUtil.handleBadRequest("Workflow status is not defined", log);
+                    String errorMessage = "Workflow status is not defined";
+                    APIMgtResourceNotFoundException e = new APIMgtResourceNotFoundException(errorMessage,
+                            ExceptionCodes.WORKFLOW_STATE_MISSING);
+                    HashMap<String, String> paramList = new HashMap<>();
```

Comment:
RestApiUtil.getErrorDTO() method accepts HashMap instead of Map, Will have to refactor RestApiUtil